pyqtorch/noise.py

```diff
@@ -0,0 +1,420 @@
+from __future__ import annotations
+
+from math import sqrt
+from typing import List, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from pyqtorch.apply import apply_ope_ope
+from pyqtorch.matrices import DEFAULT_MATRIX_DTYPE, IMAT, XMAT, YMAT, ZMAT, _dagger
+from pyqtorch.utils import density_mat
+
+
+class Noise(torch.nn.Module):
+    def __init__(
+        self, kraus: List[Tensor], target: int, probability: Union[Tuple[float, ...], float]
+    ) -> None:
+        super().__init__()
+        self.target: int = target
+        self.qubit_support: Tuple[int, ...] = (self.target,)
+        # Verification Kraus operators:
+        if not isinstance(kraus, list) or not all(isinstance(tensor, Tensor) for tensor in kraus):
+            raise TypeError("The Kraus operators must be a list containing Tensor objects")
+        if len(kraus) == 0:
+            raise ValueError("The noisy gate must be described by the Kraus operator")
+        # Create a buffer for the Kraus operators.
+        self.kraus = []
+        for index, tensor in enumerate(kraus):
+            # Registering the tensor as a buffer with a unique name
+            self.register_buffer(f"kraus_{index}", tensor)
+            # Storing the tensor itself in the self.kraus list
```

Comment:
Be careful as not to state the obvious in comments. Comments IMHO should describe points that are not expressible in code itself otherwise it makes for an unnecessary line.

Reply:
Noted.